generator/booking_dataset.cpp

```diff
@@ -161,64 +155,71 @@ vector<size_t> BookingDataset::GetNearestHotels(double lat, double lon, size_t l
   for_each(bgi::qbegin(m_rtree, bgi::nearest(TPoint(lat, lon), limit)), bgi::qend(m_rtree),
            [&](TValue const & v)
            {
-             auto const & hotel = m_hotels[v.second];
+             auto const & hotel = GetHotel(v.second);
              double const dist = ms::DistanceOnEarth(lat, lon, hotel.lat, hotel.lon);
              if (maxDistance != 0.0 && dist > maxDistance /* max distance in meters */)
                return;
 
              indexes.emplace_back(v.second);
            });
+
   return indexes;
 }
 
-void BookingDataset::BuildFeatures(function<void(OsmElement *)> const & fn) const
+void BookingDataset::BuildFeature(size_t const hotelIndex,
+                                  function<void(FeatureBuilder1 &)> const & fn) const
 {
-  for (auto const & hotel : m_hotels)
-  {
-    OsmElement e;
-    e.type = OsmElement::EntityType::Node;
-    e.id = 1;
-
-    e.lat = hotel.lat;
-    e.lon = hotel.lon;
-
-    e.AddTag("sponsored", "booking");
-    e.AddTag("name", hotel.name);
-    e.AddTag("ref:sponsored", strings::to_string(hotel.id));
-    e.AddTag("website", hotel.descUrl);
-    e.AddTag("rating:sponsored", strings::to_string(hotel.ratingUser));
-    e.AddTag("stars", strings::to_string(hotel.stars));
-    e.AddTag("price_rate", strings::to_string(hotel.priceCategory));
-    e.AddTag("addr:full", hotel.address);
-
-    if (!hotel.translations.empty())
-    {
-      vector<string> parts;
-      strings::ParseCSVRow(hotel.translations, '|', parts);
-      CHECK(parts.size() % 3 == 0, ());
-      for (auto i = 0; i < parts.size(); i += 3)
-      {
-        e.AddTag("name:" + parts[i], parts[i + 1]);
-        e.AddTag("addr:full:" + parts[i], parts[i + 2]);
-      }
-    }
+  auto const & hotel = m_hotels[hotelIndex];
+
+  FeatureBuilder1 bookingFb;
```

Comment:
s/bookingFb/fb/g
There are no other feature builders. Usage of the word "booking" in the BookingDataset methods is redundant.